lib/Dialect/FIRRTL/Ops.cpp

```diff
@@ -58,6 +59,34 @@ static ParseResult parseCircuitOp(OpAsmParser &parser, OperationState &result) {
   return success();
 }
 
+static LogicalResult verifyCircuitOp(CircuitOp &circuit) {
+  StringRef main = circuit.name();
+
+  // Check that the circuit must have a "main" module (top module)
+  if (main.empty()) {
+    circuit.emitOpError();
```

Comment:
Please add a message to the error.  Also (if reasonable), it would be great to add a testcase to `test/FIRParser/errors.fir` that triggers this.

Reply:
Message added and test included.